### PR TITLE
Enhancements to fuzz-rasm2

### PR DESCRIPTION
**Pull request by SchumBlubBlub**
- Fixed Synchronization broken when switched to ProcessPool,  No more dupes
- Added one more level of canonical so quick filters out equivalent cases

cc @Svenito 

metametametametametametametametametametametametametametametametameta... :metal:

### CODE REVIEW

File: fuzz-scripts/fuzz_rasm2.py [hunk 1]
```diff
@@ -116,7 +155,18 @@ def main():
...
+                continue
+            inskey = res['case']
+            insmkey = res['metacase']
+            insmmkey = res['metametacase']
+            meta_meta_cases[insmmkey] = meta_meta_cases.get(insmmkey, 0) + 1
+            meta_cases[insmkey] = meta_cases.get(insmkey, 0) + 1
+            if (meta_cases[insmkey] > MAX_METACASE_EXAMPLES or
+                    meta_meta_cases[insmmkey] > MAX_META_META_CASE_EXAMPLES):
+                pass
+            elif(inskey not in cases):
```
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: () not needed here

File: fuzz-scripts/fuzz_rasm2.py [hunk 2]
```diff
@@ -116,7 +155,18 @@ def main():
...
-            if (res):
+            if not res:
+                continue
+            inskey = res['case']
+            insmkey = res['metacase']
+            insmmkey = res['metametacase']
+            meta_meta_cases[insmmkey] = meta_meta_cases.get(insmmkey, 0) + 1
+            meta_cases[insmkey] = meta_cases.get(insmkey, 0) + 1
+            if (meta_cases[insmkey] > MAX_METACASE_EXAMPLES or
+                    meta_meta_cases[insmmkey] > MAX_META_META_CASE_EXAMPLES):
```
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: fix indent

File: fuzz-scripts/fuzz_rasm2.py [hunk 3]
```diff
@@ -66,14 +100,15 @@ def check_hexpairs(orig_input_hexpairs):
 
     if input_ins == 'invalid':
         if REFERENCE_DISASM_ENGINE:
-            coutput = rasm2("-D", "-a", REFERENCE_DISASM_ENGINE, orig_input_hexpairs)
+            coutput = rasm2(
+                "-D", "-a", REFERENCE_DISASM_ENGINE, orig_input_hexpairs)
             coutput = coutput.stdout.split(b"\n")[0].decode()
             coutput = re.split("\s+", coutput, 2)[-1]
             cinput_hexpairs, cinput_ins = re.split("\s+", coutput, 1)
             if (cinput_ins != 'invalid'):
```
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: Don't need () here

File: fuzz-scripts/fuzz_rasm2.py [hunk 4]
```diff
@@ -116,7 +155,18 @@ def main():
                   for o in range(offset, offset+CONCURRENCY)]
```
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: shouldn't this be `:` not `]` at the EOL

File: fuzz-scripts/fuzz_rasm2.py [hunk 5]
```diff
@@ -91,7 +126,11 @@ def check_hexpairs(orig_input_hexpairs):
         return gen_testcase("Assemble != Dis+Assemble", input_ins,
                             input_hexpairs, output_ins)
 
+
 def main():
+    cases = {}
+    meta_cases = {}
+    meta_meta_cases = {}
 
     if (len(sys.argv) != 2):
```
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: no need for ()

File: fuzz-scripts/fuzz_rasm2.py (hunk 4 above)
Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: part of comprehension, look at context.

Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: Ah ok, in the diff it looks like a single statement

File: fuzz-scripts/fuzz_rasm2.py (hunk 2 above)
Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: that's what my auto pep8 tool does.

Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: ok, leave it then

File: fuzz-scripts/fuzz_rasm2.py (hunk 3 above)
Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: Yeah, but I've spent all day in c. 

I know this is normal in r2land, but I wish code review was not just 15 comments about whitepsace every time.
something which can be automated.

Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: ok, just pointers, not the end of the world

Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: Tell me if it's useful to you or not instead, of if something is broken or missing, or could be better...

File: fuzz-scripts/fuzz_rasm2.py (hunk 4 above)
Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: I know, used to be a way to expand diff context in GH, can't find it

File: fuzz-scripts/fuzz_rasm2.py (hunk 3 above)
Comment (Svenito) [inline fuzz-scripts/fuzz_rasm2.py]: well, lgtm otherwise. I'll merge

Comment (SchumBlubBlub) [inline fuzz-scripts/fuzz_rasm2.py]: I force pushed all the fixes in any case. Just complaining...

**Pull request merged**